src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -684,7 +690,19 @@ struct MinimumSpanningTree
 #else
         false;
 #endif
-    if (use_lower_bounds)
+
+    // Shared radii may or may not be faster for CUDA depending on the problem.
+    // In the ICPP'51 paper experiments, we ended up using it only in Serial.
+    // But we would like to keep an option open for the future, so the code is
+    // written to be able to run it if we want.
+    constexpr bool use_shared_radii =
+#ifdef KOKKOS_ENABLE_SERIAL
+        std::is_same<ExecutionSpace, Kokkos::Serial>::value;
```

Comment:
In `Serial` you don't need `atomics` anyway, right? It should be enough to just use a reference instead of a copy.

Reply:
We decided to go with boolean (shared+atomic, non-shared) instead of tri-state (shared atomic, shared non-atomic, non-shared) for the code to be a bit simpler.

Reply:
> In `Serial` you don't need `atomics` anyway, right? It should be enough to just use a reference instead of a copy.

Why would we use a reference?!

Reply:
The code would be even simpler with the same effect if you dropped `atomics` here altogether (which only works for the `Serial` backend, of course).

Reply:
That would become a bug if we attempt to use shared radii with other backend